extensions/oplogPopulator/modules/Connector.js

```diff
@@ -253,14 +281,23 @@ class Connector {
      * Connector is updated with new bucket list
      * @param {string} bucket bucket to add
      * @param {boolean} [doUpdate=false] updates connector if true
-     * @returns {Promise|undefined} undefined
+     * @returns {Promise<boolean>} true if bucket was removed from
+     * the bucket
      * @throws {InternalError}
      */
     async removeBucket(bucket, doUpdate = false) {
         this._buckets.delete(bucket);
+        // Here, we must avoid removing the bucket from the
+        // bucketsInUse set, as the connector might still be
+        // using it in a deployed pipeline.
         this._updateConnectorState(true);
         try {
+            if (this._isImmutable()) {
```

Comment:
seems to break abstraction, this is not the role of the `Connector`...

Allocator/... decides what to do, and the Connector should just do it